### PR TITLE
Major refactoring

### PR DESCRIPTION
**Pull request by dev-rke**
+ migrated controllers to captcha middleware
+ removed hook
+ dropped cache integration
+ use user session only for storing captcha phrases
+ added captcha lifetime option to typoscript
+ implemented new refresh button logic

should solve #18, #19, #11

### CODE REVIEW

File: Classes/Middleware/Captcha.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+class Captcha implements MiddlewareInterface
+{
+    protected ResponseFactoryInterface $responseFactory;
+    protected ConfigurationManager $configurationManager;
+    protected TypoScriptFrontendController $tsfe;
+
+    public function __construct(
+        ResponseFactoryInterface $responseFactory,
+        ConfigurationManager $configurationManager,
+        TypoScriptFrontendController $tsfe,
```
Comment (EvilBMP) [inline Classes/Middleware/Captcha.php:26]: PHP parse error, because of comma at the end

File: Configuration/TypoScript/setup.typoscript [hunk 2]
```diff
@@ -25,37 +25,36 @@ plugin.tx_form {
 	)
 }
 
-plugin.tx_bwcaptcha {
+module.tx_bwcaptcha {
```
Comment (EvilBMP) [inline Configuration/TypoScript/setup.typoscript:28]: Why are you switching from plugin to module? The module key is normaly used for backend relevant TypoScript.

Comment (dev-rke) [reply Configuration/TypoScript/setup.typoscript:28]: You are right, mostly for backend but also for non-plugin TypoScript like middlewares or CLI applications.
So you just changed "module" back to "plugin" in TS?
Actually that should then fall back to default settings and not respect your typoscript configuration.
My middleware only parses configs from "module".

File: Classes/Middleware/Captcha.php (hunk 1 above)
Comment (dev-rke) [reply Classes/Middleware/Captcha.php]: fixed.

File: Configuration/TypoScript/setup.typoscript (hunk 2 above)
Comment (EvilBMP) [reply Configuration/TypoScript/setup.typoscript:28]: That's why I changed the identifier back to plugin in your middleware as well. The middleware is called via an asynchronous request - so we are still in the frontend. The project I tested your pull request in (TYPO3 v11.5.24), the middleware is able to read the TypoScript data from plugin.tx_bwcaptcha...

And yes, my custom TypoSript settings and not just the default values. Have you tested my pull request?

File: Classes/Middleware/Captcha.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+class Captcha implements MiddlewareInterface
+{
+    protected ResponseFactoryInterface $responseFactory;
+    protected ConfigurationManager $configurationManager;
+    protected TypoScriptFrontendController $tsfe;
+
+    public function __construct(
+        ResponseFactoryInterface $responseFactory,
+        ConfigurationManager $configurationManager,
+        TypoScriptFrontendController $tsfe
```
Comment (maikschneider) [inline Classes/Middleware/Captcha.php:26]: In v12 the `TypoScriptFrontendController` cannot be loaded via DI anymore ([Changelog](https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Breaking-96107-DeprecatedFunctionalityRemoved.html)) - so this breaks the whole page.
Accessing the user session via globals `$GLOBALS['TSFE']->fe_user` still works.

Comment (dev-rke) [reply Classes/Middleware/Captcha.php:26]: Thanks, good point. I will change this.

Comment (maikschneider) [reply Classes/Middleware/Captcha.php:26]: Hey @dev-rke, how is it going? Looking forward to merge your PR :) 
Just let me know if you don't have time right know - I can also do these little adjustments.

**Pull request merged**